src/plugins/visualizations/server/plugin.ts

```diff
@@ -77,6 +81,17 @@ export class VisualizationsPlugin
         category: ['visualization'],
         schema: schema.boolean(),
       },
+      [VISUALIZE_DISABLE_BUCKET_AGG_SETTING]: {
+        name: i18n.translate('visualizations.advancedSettings.visualizeDisableBucketAgg', {
+          defaultMessage: 'Disable visualizations bucket aggregation types',
+        }),
+        value: [],
+        description: i18n.translate('visualizations.advancedSettings.visualizeDisableBucketAgg', {
+          defaultMessage: `only visualizations' bucket aggregations that are not disabled will be available to the user.`,
```

Comment:
NIT: 
Perhaps @kgcreative could provide some comments on the wording.

It's not immediately clear what value is accept here. It's a CSV, in other CSV settings it will start with `A comma-separated list`. So I'd imagine something on that line. Also, it will take the value of the constant, do we need some [docs](https://github.com/opensearch-project/documentation-website/issues/new?assignees=&labels=proposal&template=PROPOSAL_TEMPLATE.md&title=%5BPROPOSAL%5D) that show the list and linked in the setting: 
```
  filter, filters, histogram, ip_range, date_range, range, terms, significant_terms, geohash_grid, geotile_grid, date_histogram
```

Reply:
Can we rephrase to something like, "Disabled visualizations bucket aggregations are not available to the user." The setting disables aggregation types, so I'm hesitant about describing the opposite of what the setting does (user can only see non-disabled aggregation types).

Reply:
Suggest changing `only visualizations' bucket aggregations that are not disabled will be available to the user.` to `Deactivates the specified bucket aggregations from visualizations`. This will be in the "Visualizations" section, so it's redundant to specify that it only applies to visualizations.

Reply:
Any comment about adding `A comma-separated list` to the message? Here is how the settings look by default:

![Screen Shot 2022-02-09 at 1 27 20 PM](https://user-images.githubusercontent.com/12060704/153292967-264a5a9c-751d-4a1e-ab98-a922362ebd4e.png)

Reply:
I like the idea of showing some sort of default or suggested values to improve the customer experience